### PR TITLE
recover assignment

### PR DESCRIPTION
**Pull request by apmccartney**

### CODE REVIEW

File: src/interpolation/table/Variant.hpp [hunk 1]
```diff
@@ -127,6 +127,10 @@ class Variant {
   Variant( Variant& ) = default;
   Variant( Variant&& ) = default;
   Variant( const Variant& ) = default;
+
+  Variant& operator=( Variant& ) = default;
```
Comment (apmccartney) [inline src/interpolation/table/Variant.hpp]: can be deleted

File: src/interpolation/table/Type/src/ctor.hpp [hunk 2]
```diff
@@ -46,4 +46,19 @@ Type( Type&& other ) :
...
   searchAlgorithm( this->xRange ){}
 
+Type& operator=( const Type& other ){
+  this->xRange = other.xRange;
+  this->yRange = other.yRange;
+  this->searchAlgorithm = SearchAlgorithm{ this->xRange };
+  return *this;
+}
+
+Type& operator=( Type&& other ){
```
Comment (apmccartney) [inline src/interpolation/table/Type/src/ctor.hpp:56]: should check whether reindexing is necessary

Comment (apmccartney) [reply src/interpolation/table/Type/src/ctor.hpp:56]: yup =(

File: src/interpolation/table/Type/src/ctor.hpp [hunk 3]
```diff
@@ -46,4 +46,19 @@ Type( Type&& other ) :
...
+  this->xRange = std::move(other.xRange);
```
Comment (jlconlin) [inline src/interpolation/table/Type/src/ctor.hpp:57]: If you are moving data members from one instance to another, does that mean that `other` becomes unusable?

**Review by jlconlin — APPROVED** [review]

File: src/interpolation/table/Type/src/ctor.hpp (hunk 3 above)
Comment (apmccartney) [reply src/interpolation/table/Type/src/ctor.hpp:57]: > The post-condition is that a moved-from object is in an unspecified, but valid state.

Or in otherwords, until other is reassigned, operations on it are undefined

**Pull request merged**